automerge-cli/src/export.rs

```diff
@@ -0,0 +1,42 @@
+use crate::error::AutomergeCliError;
+use automerge_backend::AutomergeError;
+use automerge_frontend::Value;
+use automerge_protocol::Patch;
+use std::fs::File;
+use std::path::Path;
+use std::{io, io::Read};
+
+fn get_changes(changes_file: &Path) -> io::Result<Vec<u8>> {
+    let mut input_data = Vec::new();
+    let mut input_file = File::open(changes_file)?;
+    input_file.read_to_end(&mut input_data)?;
+
+    Ok(input_data)
+}
+
+fn get_patch(changes_bytes: Vec<u8>) -> Result<Patch, AutomergeError> {
+    let mut backend = automerge_backend::Backend::init();
+    backend.load_changes_binary(vec![changes_bytes]).unwrap();
+
+    backend.get_patch()
+}
+
+fn get_state(changes_file: &Path) -> Result<Value, AutomergeCliError> {
+    let input_data =
+        get_changes(changes_file).map_err(|_| AutomergeCliError::InvalidChangesFile)?;
+
+    let patch = get_patch(input_data).map_err(|_| AutomergeCliError::BackendError)?;
+    let mut frontend = automerge_frontend::Frontend::new();
+    frontend.apply_patch(patch).unwrap();
+
+    Ok(frontend.state().clone())
```

Comment:
Is there a way to avoid the `clone` call here?

Reply:
Switched this to return a `serde::Value` here since that's ultimately what we want to use in the consumer, but we'll need to update this once we need to export to other formats so it'd be good to know what the generally accepted pattern here is: we don't really need to keep the `frontend` around, so how should we return the `state` and release `frontend`? Or is that not idiomatic at all and we should return the `frontend` and use it to get the state in the consuming function?

Reply:
Hmm, there are a bunch of different approaches. The easiest way is probably to add a consuming method to the frontend which returns the state and drops the rest. I think we can probably punt on this until we have more understanding of what the usecases for the CLI look like though.